### PR TITLE
[BACKEND] B-email-verification

### LINKED ISSUE #252: [BACKEND] Email verification
Email verification issue is necessary for verify user account. Usage of specific gmail account will be satisfied with given ENV variables later.

### LINKED ISSUE #304: [BACKEND] Set .env variable
ENV variables should be added repo and then gitignored

### PR DESCRIPTION
**Pull request by bekir96**
## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Description

I have implemented two endpoints to verify mail account. Collaborated with @abdullahyldz on frontend side.

First endpoint is `email-verify/<uidb64>`. This endpoint firstly come up with register side. When customer register to own page, our gmail account send verify link to customer mail. Verify link contains token in  path, and token expire date is 2 days.  

Second endpoint is `user/verify`. This endpoint come up with profile page. When customer enters own profile page if do not verify your account, customer see verify your account section. Then clicks the this section, our gmail account send verify link to customer mail again.

To test our endpoints with @abdullahyldz , he helps me to run frontend side. To test locally, backend is starting at `http://127.0.0.1:1234/` and frontend is starting at `http://127.0.0.1:5000/`.

# Screenshots

image
WhatsApp Image 2020-12-22 at 22 28 36
WhatsApp Image 2020-12-22 at 22 29 05
image
image
image

# Test results

Test results are satisfied above Screenshot section.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have checked my code and corrected any misspellings
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# Potential problems/Important remarks

All ENV variables are given by Slack.

### CODE REVIEW

**Review by omerfdeniz — APPROVED** [review]
Good job, can merge

**Review by abdullahyldz — APPROVED** [review]
We have manually tested with @bekir96 and verified the correctness of code.
Great job.

File: ecommerce/backend/getflix/settings.py [hunk 1]
```diff
@@ -26,9 +26,14 @@
     'HOST': os.environ.get('DB_HOST'),
     'USER': os.environ.get('DB_USER'),
     'PASSWORD': os.environ.get('DB_PASSWORD'),
+    'EMAIL_BACKEND': os.environ.get('EMAIL_BACKEND'),
+    'MAILER_EMAIL_BACKEND': os.environ.get('MAILER_EMAIL_BACKEND'),
+    'EMAIL_HOST': os.environ.get('EMAIL_HOST'),
+    'EMAIL_HOST_PASSWORD': os.environ.get('EMAIL_HOST_PASSWORD'),
+    'EMAIL_HOST_USER': os.environ.get('EMAIL_HOST_USER'),
+    'DEFAULT_FROM_EMAIL': os.environ.get('DEFAULT_FROM_EMAIL')
```
Comment (umutoksuz) [inline ecommerce/backend/getflix/settings.py]: For those, I think we need to configure our backend deploy script and add encrypted version of email password etc. to repo.

File: ecommerce/backend/getflix/settings.py [hunk 2]
```diff
@@ -48,6 +53,19 @@ def get_secret(setting, secrets=secrets):
     except KeyError:
         raise ImproperlyConfigured("Set the {} setting".format(setting))
 
+EMAIL_BACKEND = get_secret('EMAIL_BACKEND')
+MAILER_EMAIL_BACKEND = get_secret('MAILER_EMAIL_BACKEND')
+EMAIL_HOST = get_secret('EMAIL_HOST')
+EMAIL_HOST_PASSWORD = get_secret('EMAIL_HOST_PASSWORD')
+EMAIL_HOST_USER = get_secret('EMAIL_HOST_USER')
```
Comment (umutoksuz) [inline ecommerce/backend/getflix/settings.py:78]: These lines are testing purpose? We should delete them before deploying to master branch

**Review by umutoksuz — COMMENTED** [review]
Code is well. About some minor points I have small comments

File: ecommerce/backend/getflix/settings.py (hunk 1 above)
Comment (umutoksuz) [reply ecommerce/backend/getflix/settings.py]: Merging to master after those variables are configured would be better for deployment purposes

File: ecommerce/backend/getflix/settings.py (hunk 2 above)
Comment (bekir96) [reply ecommerce/backend/getflix/settings.py:78]: No, ENV variables are:
export EMAIL_BACKEND="django.core.mail.backends.smtp.EmailBackend"
export MAILER_EMAIL_BACKEND=$EMAIL_BACKEND
export EMAIL_HOST="smtp.gmail.com"
export EMAIL_HOST_PASSWORD="getflixecommerce"
export EMAIL_HOST_USER="getflix.e.commerce"
export DEFAULT_FROM_EMAIL=$EMAIL_HOST_USER

group should add these to env variables

**Comment by MehdiSaffar** [comment]
added the env handling part, will need to be tested tomorrow before merging.

@bekir96 has the .env.production and .env.development files that need to be put in `ecommerce/backend`. 

During dev, for DB_HOST you can pass it either as env variable from command line or by putting it in `.env.development`

Example on linux:

```
DB_HOST=111.222.333.444 python manage.py runserver
```

Please do not make any changes to settings.py from now on

**Review by MehdiSaffar — APPROVED** [review]
Integrated necessary changes, works perfectly (tested locally and deployed frontend)

**Pull request merged**